libs/src/api/challenge.js

```diff
@@ -0,0 +1,310 @@
+        specifier,
+        recipientEthAddress,
+        oracleEthAddress,
+        tokenAmount: fullTokenAmount
+      })
+
+      return { success: true, error: false }
+    } catch (e) {
+      const err = e.message
+      console.log(`Failed to submit and evaluate attestations at phase ${phase}: ${err}`)
+      return { success: false, error: e }
+    }
+  }
+
+  /**
+   *
+   * Aggregates attestations from Discovery Nodes and AAO.
+   *
+   * @typedef {Object} AttestationsReturn
+   * @property {Array<AttestationMeta>} discoveryNodeAttestations
+   * @property {AttestationMeta} aaoAttestation
+   *
+   * @param {{
+   *   challengeId,
+   *   encodedUserId,
+   *   handle,
+   *   specifier,
+   *   oracleEthAddress,
+   *   amount,
+   *   quorumSize,
+   *   AAOEndpoint,
+   *   endpoints = null
+   * }} {
+   *   challengeId,
+   *   encodedUserId,
+   *   handle,
+   *   specifier,
+   *   oracleEthAddress,
+   *   amount,
+   *   quorumSize,
+   *   AAOEndpoint,
+   *   endpoints = null
+   * }
+   * @returns {Promise<AttestationsReturn>} attestations
+   * @memberof Challenge
+   */
+  async aggregateAttestations ({ challengeId, encodedUserId, handle, specifier, oracleEthAddress, amount, quorumSize, AAOEndpoint, endpoints = null }) {
+    this.REQUIRES(Services.DISCOVERY_PROVIDER)
+
+    // If no endpoints array provided, select here
+    if (!endpoints) {
+      endpoints = await this.discoveryProvider.serviceSelector.findAll()
+    }
+    endpoints = sampleSize(endpoints, quorumSize)
+
+    try {
+      const discprovAttestations = endpoints.map(e => this.getChallengeAttestation({ challengeId, encodedUserId, specifier, oracleEthAddress, discoveryProviderEndpoint: e }))
+      const AAOAttestation = this.getAAOAttestation({
+        challengeId,
+        specifier,
+        handle,
+        amount,
+        AAOEndpoint,
+        oracleEthAddress
+      })
+
+      const res = await Promise.all([...discprovAttestations, AAOAttestation])
+      const discoveryNodeAttestationResults = res.slice(0, -1)
+      const discoveryNodeAttestations = discoveryNodeAttestationResults.map(r => r.success)
+      const discoveryNodeAttestationErrors = discoveryNodeAttestationResults.map(r => r.error)
+      const { success: aaoAttestation, error: aaoAttestationError } = res[res.length - 1]
+      console.log({ discoveryNodeAttestations, aaoAttestation })
+
+      const error = aaoAttestationError || discoveryNodeAttestationErrors.find(Boolean)
+      if (error) {
+        return {
+          discoveryNodeAttestations: null,
+          aaoAttestation: null,
+          error
+        }
+      }
+
+      return {
+        discoveryNodeAttestations,
+        aaoAttestation,
+        error: null
+      }
+    } catch (e) {
+      const err = e.message
+      console.log(`Failed to aggregate attestations: ${err}`)
+      return {
+        discoveryNodeAttestations: null,
+        aaoAttestation: null,
+        error: GetAttestationError.UNKNOWN_ERROR
```

Comment:
console.error instead of log? i think it'll make it stick around in sentry longer, etc.